modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,260 @@
+
+.JSON Schema (`example/schema/Person.json`):
+[source,JSON]
+----
+{
+  "$id": "https://example.com/person.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Person",
+  "type": "object",
+  "properties": {
+    "firstName": {
+      "type": "string",
+      "description": "The person's first name."
+    },
+    "lastName": {
+      "type": "string",
+      "description": "The person's last name."
+    },
+    "age": {
+      "description": "Age in years which must be equal to or greater than zero.",
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "required": ["firstName", "lastName"]
+}
+----
+
+The following example shows how you can use the JSON schema loader to import the type:
+
+.DataWeave Script Header:
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Import Syntax
+
+To import JSON schema types, use the following syntax, where:
+
+* `typeToImport`: You can use `*` alone to import all types defined in the schema or you can import a single type from the schema using, for example, `Root`. You can also import JSON schema types with a different name, for example, `Root as Person`, which enables you to reference the type with that name in the script.
+* `pathToJsonSchema`: To specify the path to the schema file, replace the file separators with `::` and remove the `.json` extension from the file name. For example, if the path to the schema is `example/schema/Person.json`, use `example::schema::Person`.
+
+----
+import _typesToImport_ from jsonschema!_pathToJsonSchema_
+----
+
+The following example shows how to import a type:
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Use Your Types in a DataWeave Script
+
+Having the import directive from the above example in the script header loads all the existing types in the JSON schema. In `import * from jsonschema!example::schema::Person`, the only existing type is the `Root` type, specified at the root which describes an Object that can have three properties (`firstName`, `lastName`, and `age`). It is equivalent to declaring the following type in your DataWeave script:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+type Root = { firstName: String, lastName: String, age?: Number }
+----
+
+Notice that the `age` field is the only one that is optional, as indicated by the `?`.
+
+You can use the type to know if a value follows the structure defined by the JSON schema.
+The following example outputs the value `true` because the object contains the required fields, `firstName` and `lastName`:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ lastName: "Doe"
+} is Root
+----
+
+.Output:
+----
+"true"
+----
+
+The following example outputs the value `false` since the object does not contain the required fields, `firstName` and `lastName`:
```

Comment:
```suggestion
The following example outputs the value `false` because the object does not contain the required fields, `firstName` and `lastName`:
```

Reply:
Use "since" for time.
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-synonymsandwordswithmultiplemeanings

Reply:
Reworded.